ert_shared/ensemble_evaluator/prefect_ensemble/client.py

```diff
@@ -0,0 +1,36 @@
+import websockets
+from websocket import WebSocketException
+import asyncio
+
+
+class Client:
+    def __enter__(self, max_retries=10):
+        self._max_retries = max_retries
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        if self.websocket is not None:
+            self.loop.run_until_complete(self.websocket.close())
+        self.loop.close()
+
+    def __init__(self, url):
+        self.url = url
+        self.websocket = None
+        self.loop = asyncio.new_event_loop()
+
+    async def get_websocket(self):
+        return await websockets.connect(self.url)
+
+    async def _send(self, msg):
+        for retry in range(self._max_retries):
+            try:
+                if self.websocket is None:
+                    self.websocket = await self.get_websocket()
+                await self.websocket.send(msg)
+                break
+            except (ConnectionRefusedError, WebSocketException):
+                await asyncio.sleep(0.2 + 5 * retry)
```

Comment:
I know I'm the author of this snippet, but maybe we should rethink the waiting strategy a bit. 5 + 10 + ... + 45 = a really large number, so maybe it makes more sense to specify a timeout instead?